wardrobe/timestamp/src/lib.rs

```diff
@@ -0,0 +1,382 @@
+        // Extract the current timestamp from the inherent data
+        let timestamp_millis: u64 = authoring_inherent_data
+            .get_data(&sp_timestamp::INHERENT_IDENTIFIER)
+            .expect("Inherent data should decode properly")
+            .expect("Timestamp inherent data should be present.");
+        let new_timestamp = Timestamp {
+            time: timestamp_millis,
+            block: T::block_height(),
+        };
+
+        log::debug!(
+            target: LOG_TARGET,
+            "🕰️🖴 Local timestamp while creating inherent i:: {timestamp_millis}"
+        );
+
+        let mut peeks = Vec::new();
+        match (previous_inherent, T::block_height()) {
+            (None, 1) => {
+                // This is the first block hack case.
+                // We don't need any inputs, so just do nothing.
+            }
+            (None, _) => panic!("Attemping to construct timestamp inherent with no previous inherent (and not block 1)."),
+            (Some((_previous_inherent, previous_id)), _) => {
+                // This is the the normal case. We create a full previous to peek at.
+
+                // We are given the entire previous inherent in case we need data from it or need to scrape the outputs.
+                // But out transactions are simple enough that we know we just need the one and only output.
+                peeks.push(OutputRef {
+                    tx_hash: previous_id,
+                    // There is always 1 output, so we know right where to find it.
+                    index: 0,
+                });
+            }
+        }
+
+        let new_output = Output {
+            payload: new_timestamp.into(),
+            verifier: UpForGrabs.into(),
+        };
+
+        Transaction {
+            inputs: Vec::new(),
+            peeks,
+            outputs: vec![new_output],
+            checker: Self::default(),
+        }
+    }
+
+    fn check_inherent(
+        importing_inherent_data: &InherentData,
+        inherent: Transaction<V, Self>,
+        result: &mut CheckInherentsResult,
+    ) {
+        // Extract the local view of time from the inherent data
+        let local_time: u64 = importing_inherent_data
+            .get_data(&sp_timestamp::INHERENT_IDENTIFIER)
+            .expect("Inherent data should decode properly")
+            .expect("Timestamp inherent data should be present.");
+
+        log::debug!(
+            target: LOG_TARGET,
+            "🕰️🖴 Local timestamp while checking inherent is: {:#?}", local_time
+        );
+
+        let on_chain_timestamp = inherent.outputs[0].payload.extract::<Timestamp>().expect(
+            "SetTimestamp extrinsic should have an output that decodes as a StorableTimestamp.",
+        );
+
+        log::debug!(
+            target: LOG_TARGET,
+            "🕰️🖴 In-block timestamp is: {:#?}", on_chain_timestamp
+        );
+
+        // Although FRAME makes the check for the minimum interval here, we don't.
```

Comment:
Again references to FRAME might not be necessary since users of Tuxedo can come from other places not caring about FRAME

Reply:
Yeah, I hear your point, and I took it on the other comment. But I do think it is a good idea to document the places where we take different design decisions than FRAME. I think it is more appropriate in a piece like this that directly corresponds to a specific frame pallet. I'm down to remove it though.